File: opensrp-chw-referral/src/main/java/org/smartregister/chw/referral/contract/BaseReferralHistoryContract.java
package org.smartregister.chw.referral.contract;

import android.content.Context;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.smartregister.chw.referral.domain.MemberObject;

import java.util.Collection;
import java.util.List;

public interface BaseReferralHistoryContract {
    interface View {
        Presenter presenter();

        Context getCurrentContext();

        void setProfileViewWithData(MemberObject memberObject);

        void setReferralHistoryData(@NotNull Collection<Object> referrals);
    }

    interface Presenter {
        void fillClientData(@Nullable MemberObject memberObject);

        @Nullable
        BaseReferralHistoryContract.View getView();

        String getMainTable();

        String getMainCondition();
    }

    interface Model {
        List<Object> getReferrals(String baseEntityId);
    }


}

File: opensrp-chw-referral/src/main/java/org/smartregister/chw/referral/contract/BaseReferralHistoryContract.kt
package org.smartregister.chw.referral.contract

import org.smartregister.chw.referral.domain.MemberObject

interface BaseReferralHistoryContract {

    interface View {

        fun presenter(): Presenter?

        fun setProfileViewWithData(memberObject: MemberObject)
    }

    interface Presenter {

        fun fillClientData(memberObject: MemberObject)

        fun getView(): View?

        fun getMainTable(): String

        fun getMainCondition(): String
    }

    interface Model {
        fun getReferrals(baseEntityId: String?): List<Any?>?
    }
}